### PR TITLE
Bump Prototype Kit and update repo

### PR DESCRIPTION
**Pull request by NickWalt01**

### CODE REVIEW

**Comment by NickWalt01** [comment]
Files to be reviewed are:

cd.yaml
dependabot.yml
README.md
License
service.yaml
deployment.tpl
ingress.yaml
Dockerfile

The remaining files are auto generated code and deleted files.

**Comment by jackstockley89** [comment]
I have added comments on the deployments files. We haven't tested the v12 update, Would you be able to test creating a prototype repository with this template and deploy within the cloud-platform

**Comment by NickWalt01** [comment]
Hi Jack, sure I will give it a go. I cannot see the comments?

File: README.md [hunk 1]
```diff
@@ -4,22 +4,32 @@
...
+* kubernetes/ingress.yaml
+* kubernetes/deployment.tpl
+
+**Local Development**
+
+Run the command npm start and open http://localhost:3000 in a web browser.
+
+**Updating Protype Kit**
+
+To update protype kit in this repository see the [update guide](https://govuk-prototype-kit.herokuapp.com/docs/updating-the-kit)
```
[the remark this answers is not in the document]
Comment (NickWalt01) [reply README.md:28]: That is a url to the instructions on how to update the auto generated code when bumping the prototype kit version. I think heroku is their hosting platform

File: kubernetes/ingress.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  name: moj-prototype-ingress
+  namespace: moj-prototype
+  annotations:
+    kubernetes.io/ingress.class: nginx
+    external-dns.alpha.kubernetes.io/set-identifier: moj-prototype-ingress-moj-prototype-green
+    external-dns.alpha.kubernetes.io/aws-weight: "100"
+spec:
+  tls:
+  - hosts:
+    - moj-prototype.apps.live.cloud-platform.service.justice.gov.uk
```
[the remark this answers is not in the document]
Comment (NickWalt01) [reply kubernetes/ingress.yaml:13]: yeah sure

**Comment by NickWalt01** [comment]
Hi @jackstockley89 

I've created this [repo](https://github.com/ministryofjustice/demo-updated-moj-template-project) to demo updating the template repository. It can be seen on CP [here](https://demo-updated-moj-template-project.apps.live.cloud-platform.service.justice.gov.uk/).

A lot of files are changed by the upgrade of protype kit to v12 automatically, this applies to the changes in the following folders: 
lib
app
gulp
docs

These files changed due to running standard --fix:
nodemon.js
listen-on-port.js
server.js
start.js
filters.js
auto-store-data.js
step-by-step-nav.js
documentation_routes.js
nodemon.js
core_filters.js
utils.js
utils.test.js

These files I've manually edited:
.github/dependabot.yml
.github/workflows/cd.yaml
Dockerfile
README.md
docs/nodemon.js

I have changed the kubernetes-deploy.tpl into separate files:
deployment.tpl
ingress.tpl
service.tpl

This is all ready for review, cheers.

**Review by poornima-krishnasamy — APPROVED** [review]

**Pull request merged**